javaparser-core/src/main/java/com/github/javaparser/ast/CompilationUnit.java

```diff
@@ -755,7 +760,7 @@ public Path getDirectory() {
          * Saves the compilation unit to its original location
          */
         public void save() {
-            save(cu -> printer.print(cu));
+            save(cu -> compilationUnit.getPrinter().print(cu));
```

Comment:
Is there a reason we want a printer per file, and not a printer per project?

Reply:
Actually there is a new pretty printer creation for each print action because of a state managed in internal class (Sourceprinter). LexicalPreservingPrinter acts as a per project printer, i didn't think it was the preferred orientation. I just tried to integrate a more flexible mechanism into the existing code (but maybe I didn't succeed) :-(